Create rule S6856 (#3462)
SonarSource/rspec: rules/S6856/java/rule.adoc

== Why is this an issue?

The `@PathVariable` annotation in Spring extracts values from the URI path and binds them to method parameters in a Spring MVC controller.
It is commonly used with `@GetMapping`, `@PostMapping`, `@PutMapping`, and `@DeleteMapping` to capture path variables from the URI.
These annotations map HTTP requests to specific handler methods in a controller.
They are part of the Spring Web module and are commonly used to define the routes for different HTTP operations in a RESTful API.

> If there is a method with a path template that includes a placeholder, such as "/api/resource/{id}", but no corresponding

Reviewer: Prefer to write all sentences on a single line, so it is easier to compare the differences afterwards.

> `@PathVariable` annotation on a method parameter is present to capture the value of the `id` path variable, Spring will not be able to bind the variable.

Reviewer: Maybe a bit difficult to understand, I would replace it with something like this :
```suggestion
but no `@PathVariable` captures the value of the `id` path variable, Spring will not be able to bind the variable.
```
or 
```suggestion
but no `@PathVariable` binds the placeholder to a method parameter, Spring will ignore the `id` path variable.
```

Reviewer 2: I took a bit different approach.
WDYT about:

> If a method has a path template containing a placeholder, like "/api/resource/{id}", and there's no @PathVariable annotation on a method parameter to capture the id path variable, Spring will disregard the id variable.

Reviewer: :+1:
